### PR TITLE
SONARKT-245 Add a project with better semantic information to the ruling

### PR DESCRIPTION
**Pull request by alban-auzeill**
The project `kotlin-language-server` has been added to `SonarSource/slang-test-sources` repository by this [commit](https://github.com/SonarSource/slang-test-sources/commit/fd74061ee2669fe8ff4a0f5f96c9a2ec8d9087a7).

I added a `kotlin` paragraph to [SonarSource/slang-test-sources/README.md](https://github.com/SonarSource/slang-test-sources/blob/master/README.md) file to document which version and how this project has been added.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,23 +41,42 @@ Then you need to switch to Java8 and run the command to generate binaries for Kt
...
 
 You can also build and run only Plugin Tests using the `plugin` property:
 
-    ./gradlew build -Pplugin --info --no-daemon -Dsonar.runtimeVersion=7.9
+    ./gradlew build -Pplugin --info --no-daemon -Dsonar.runtimeVersion=9.7.1.62043
 
 To run e.g. the ruling tests in the IDE, create a new Run/Debug Configuration where you run the following:
 
-    :its:ruling:test -Pruling -Dsonar.runtimeVersion=7.9
+    ./gradlew :its:ruling:test --info -Pruling
```
Comment (johann-beleites-sonarsource) [inline README.md:56]: This line describes how to configure an IDE run to execute the ruling tests - so the former version without `./gradlew` was correct

File: README.md [hunk 2]
```diff
@@ -41,23 +41,42 @@ Then you need to switch to Java8 and run the command to generate binaries for Kt
...
 
 You can also run single ruling tests, e.g.:
 
-    :its:ruling:test --tests "org.sonarsource.slang.SlangRulingTest.test_kotlin_corda" -Pruling -Dsonar.runtimeVersion=7.9
+    ./gradlew :its:ruling:test --info -Pruling --tests "org.sonarsource.slang.SlangRulingTest.test_kotlin_corda"
```
Comment (johann-beleites-sonarsource) [inline README.md:60]: Same here, we are talking about how to configure the IDE, not how to use gradle directly

File: README.md [hunk 3]
```diff
@@ -41,23 +41,42 @@ Then you need to switch to Java8 and run the command to generate binaries for Kt
...
+
+**Additional ruling parameters**
+
+* By default, the SonarQube version used is LATEST_RELEASE, you can use the following property to set a different one:
+
+  -Dsonar.runtimeVersion=9.7.1.62043
```
Comment (johann-beleites-sonarsource) [inline README.md:66]: Since we have this property documented here explicitly now, maybe we can remove it from even more of the examples above?

**Review by johann-beleites-sonarsource — COMMENTED** [review]
Nice! Only a couple of remarks in the README, otherwise LGTM!

File: its/ruling/src/test/resources/expected/kotlin/kotlin-language-server/kotlin-S1192.json [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
+'kotlin-kotlin-language-server-project:server/src/main/kotlin/org/javacs/kt/KotlinTextDocumentService.kt':[
+107,
```
Comment (alban-auzeill) [inline its/ruling/src/test/resources/expected/kotlin/kotlin-language-server/kotlin-S1192.json:3]: I created [SONARKT-248](https://sonarsource.atlassian.net/browse/SONARKT-248) to fix this false-positive.

File: its/ruling/src/test/resources/expected/kotlin/kotlin-language-server/kotlin-S5612.json [hunk 5]
```diff
@@ -0,0 +1,15 @@
+{
+'kotlin-kotlin-language-server-project:server/src/main/kotlin/org/javacs/kt/KotlinLanguageServer.kt':[
+69,
```
Comment (alban-auzeill) [inline its/ruling/src/test/resources/expected/kotlin/kotlin-language-server/kotlin-S5612.json:3]: I created [SONARKT-249](https://sonarsource.atlassian.net/browse/SONARKT-249) to fix this false-positive.

File: its/ruling/src/test/resources/expected/kotlin/kotlin-language-server/kotlin-S1186.json [hunk 6]
```diff
@@ -0,0 +1,16 @@
+{
+'kotlin-kotlin-language-server-project:server/src/main/kotlin/org/javacs/kt/KotlinLanguageServer.kt':[
+164,
```
Comment (alban-auzeill) [inline its/ruling/src/test/resources/expected/kotlin/kotlin-language-server/kotlin-S1186.json:3]: I created [SONARKT-250](https://sonarsource.atlassian.net/browse/SONARKT-250) to fix this false-positive

File: its/ruling/src/test/resources/expected/kotlin/kotlin-language-server/kotlin-S1874.json [hunk 7]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (alban-auzeill) [inline its/ruling/src/test/resources/expected/kotlin/kotlin-language-server/kotlin-S1874.json:1]: I created [SONARKT-251](https://sonarsource.atlassian.net/browse/SONARKT-251) to fix FP

File: README.md (hunk 1 above)
Comment (alban-auzeill) [reply README.md]: Note: I fixed the review cchanges part of the new [4th commit](https://github.com/SonarSource/sonar-kotlin/pull/228/commits/6183ff343d60ce579a17a1bd38c4278bd691ee66)

Comment (alban-auzeill) [reply README.md]: @johann-beleites-sonarsource I also added 4 comments on this PR with links to Jira FP tickets that I have created. Could you also review the Jira tickets to confirm they make sense?

File: its/ruling/src/test/resources/expected/kotlin/kotlin-language-server/kotlin-S1186.json (hunk 6 above)
Comment (johann-beleites-sonarsource) [reply its/ruling/src/test/resources/expected/kotlin/kotlin-language-server/kotlin-S1186.json:3]: I left a comment on this one, as the RSPEC explicitly addresses this case already, indicating that this is a TP.

**Review by johann-beleites-sonarsource — APPROVED** [review]
LGTM! I think the tickets are mostly good too, there is one ([SONARKT-250](https://sonarsource.atlassian.net/browse/SONARKT-250)) that we should discuss. Shouldn't stop us from merging this, though.

**Pull request merged**